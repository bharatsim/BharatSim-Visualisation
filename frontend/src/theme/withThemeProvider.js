import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
