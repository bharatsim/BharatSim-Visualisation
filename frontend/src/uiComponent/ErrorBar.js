import Box from '@material-ui/core/Box';
import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import ErrorButton from './ErrorButton';

const useStyles = makeStyles((theme) => ({
  errorTitle: {
    fontWeight: theme.typography.fontWeightBold,
    lineHeight: 1,
    color: theme.palette.error.dark,
  },
  errorMessage: {
    lineHeight: 1,
    color: theme.palette.error.dark,
  },
  hidden: {
    visibility: 'hidden',
    height: theme.spacing(8),
  },
  errorBox: {
    display: 'flex',
    minHeight: theme.spacing(8),
    border: '1px solid',
    borderRadius: theme.spacing(1),
    borderColor: '#FFC5B3',
    backgroundColor: '#FFEEE8',
    padding: theme.spacing(2, 6),
    justifyContent: 'space-between',
    alignItems: 'center',
    width: ({ fullWidth }) => (fullWidth ? '100%' : 'fit-content'),
  },
}));

function ErrorBar({ visible, message, errorAction, fullWidth }) {
  const classes = useStyles({ fullWidth });
  return (
    <Box className={visible ? classes.errorBox : classes.hidden}>
      <Box display="flex">
        <Typography variant="body2" color="error" classes={{ body2: classes.errorTitle }}>
          Error:&nbsp;
        </Typography>
        <Typography variant="body2" color="error" classes={{ body2: classes.errorMessage }}>
          {message}
        </Typography>
      </Box>
      {errorAction && (
        <Box ml={7}>
          <ErrorButton onClick={errorAction.onClick} size="small" variant="text">
            {errorAction.name}
          </ErrorButton>
        </Box>
      )}
    </Box>
  );
}

ErrorBar.propTypes = {
  visible: PropTypes.bool,
  message: PropTypes.string,
  fullWidth: PropTypes.bool,
  errorAction: PropTypes.shape({
    name: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
  }),
};
ErrorBar.defaultProps = {
  visible: false,
  message: '',
  errorAction: null,
  fullWidth: true,
};
export default ErrorBar;
