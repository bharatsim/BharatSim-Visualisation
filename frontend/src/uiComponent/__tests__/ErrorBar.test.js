import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import withThemeProvider from '../../theme/withThemeProvider';
import ErrorBar from '../ErrorBar';

const ErrorBarWithProvider = withThemeProvider(ErrorBar);

describe('ErrorBar', () => {
  it('should match snapshot', () => {
    const { container } = render(<ErrorBarWithProvider visible message="error message" />);

    expect(container).toMatchSnapshot();
  });

  it('should call onClick of error button', () => {
    const onClick = jest.fn();
    const { getByText } = render(
      <ErrorBarWithProvider visible message="error text" errorAction={{ name: 'test', onClick }} />,
    );

    const button = getByText('test').closest('button');

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalled();
  });
});
