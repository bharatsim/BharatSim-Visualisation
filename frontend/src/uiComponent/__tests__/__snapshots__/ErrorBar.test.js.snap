// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBar should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-5 makeStyles-errorBox-4"
  >
    <div
      class="MuiBox-root MuiBox-root-6"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 makeStyles-errorTitle-1 MuiTypography-colorError"
      >
        Error:Â 
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 makeStyles-errorMessage-2 MuiTypography-colorError"
      >
        error message
      </p>
    </div>
  </div>
</div>
`;
